cupy/cuda/stream.pyx

```diff
@@ -173,24 +190,27 @@ class Stream(object):
         return self.ptr == other.ptr
 
     def __enter__(self):
-        if not hasattr(_thread_local, 'prev_stream_ref_stack'):
-            _thread_local.prev_stream_ref_stack = []
-        prev_stream_ref = weakref.ref(get_current_stream())
-        _thread_local.prev_stream_ref_stack.append(prev_stream_ref)
-        _set_current_stream(self)
+        tls = _StreamThreadLocal.get()
+        if tls.prev_stream_ref_stack is None:
+            tls.prev_stream_ref_stack = []
+        prev_stream_ref = weakref.ref(tls.get_current_stream())
+        tls.prev_stream_ref_stack.append(prev_stream_ref)
+        tls.set_current_stream(self)
         return self
 
     def __exit__(self, *args):
-        prev_stream_ref = _thread_local.prev_stream_ref_stack.pop()
-        _set_current_stream(prev_stream_ref())
+        tls = _StreamThreadLocal.get()
+        prev_stream_ref = tls.prev_stream_ref_stack.pop()
+        tls.set_current_stream(prev_stream_ref())
```

Comment:
Similarly as above, here a strong ref is created from a weak ref and then a new weak ref is created from it. Can we remove the redundancy?